pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -122,18 +120,28 @@ in buildPythonPackage rec {
     repo   = "pytorch";
     rev    = "v${version}";
     fetchSubmodules = true;
-    sha256 = "1biyq2p48chakf2xw7hazzqmr5ps1nx475ql8vkmxjg5zaa071cz";
+    sha256 = "07ga9806c5q02hsdybwdw9nc0r1a8dinwjakyg6wlrccclc027lq";
   };
 
-  dontUseCmakeConfigure = true;
-
   preConfigure = lib.optionalString cudaSupport ''
     export TORCH_CUDA_ARCH_LIST="${lib.strings.concatStringsSep ";" final_cudaArchList}"
     export CC=${cudatoolkit.cc}/bin/gcc CXX=${cudatoolkit.cc}/bin/g++
   '' + lib.optionalString (cudaSupport && cudnn != null) ''
     export CUDNN_INCLUDE_DIR=${cudnn}/include
   '';
 
+  # Use pytorch's custom configurations
+  dontUseCmakeConfigure = true;
+
+  BUILD_NAMEDTENSOR = true;
+  BUILD_DOCS = buildDocs;
+  USE_MKLDNN = mklSupport;
+  preBuild = with lib.strings; concatStringsSep "\n" [
+    ''export CMAKE_PREFIX_PATH="$CMAKE_PREFIX_PATH:${if mklSupport then mkl else openblas}"''
```

Comment:
Shouldn't cmake's setupHook do this allready? Just add `mkl` or `openblas` to the `buildInputs`.

Reply:
Unfortunately, as you noted in https://github.com/NixOS/nixpkgs/pull/75827#discussion_r360659577, pytorch has it's own custom cmake configuration tooling.

From what I observed, you couldn't just add `mkl` or `openblas` to the buildinputs for pytorch. It will build, but the dependencies won't be hooked in and you will only observe failures during test time.

Reply:
I am still not convinced. We do set the CMAKE_PREFIX_PATH:
https://github.com/NixOS/nixpkgs/blob/bc555a24afbe0be95a90aee6cfccf626f4d6ac08/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L123
https://github.com/NixOS/nixpkgs/blob/bc555a24afbe0be95a90aee6cfccf626f4d6ac08/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L1-L3
You might need to move `(if mklSupport then mkl else openblas)` from `nativeBuildInputs` to `buildInputs` to make it work.

Reply:
You're absolutely correct and this did work out -- I guess I was expecting nativeBuildInputs to be added to the cmake path